test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java

```diff
@@ -1240,24 +1240,30 @@ public final void testSyntheticEmptyListNoDocValuesLoader() throws IOException {
     }
 
     public final void testBlockLoaderFromColumnReader() throws IOException {
-        testBlockLoader(true);
+        testBlockLoader(false, true);
     }
 
     public final void testBlockLoaderFromRowStrideReader() throws IOException {
-        testBlockLoader(false);
+        testBlockLoader(false, false);
+    }
+
+    public final void testBlockLoaderFromColumnReaderWithSyntheticSource() throws IOException {
+        testBlockLoader(true, true);
+    }
+
+    public final void testBlockLoaderFromRowStrideReaderWithSyntheticSource() throws IOException {
+        testBlockLoader(true, false);
     }
 
     protected boolean supportsColumnAtATimeReader(MapperService mapper, MappedFieldType ft) {
         return ft.hasDocValues();
     }
 
-    private void testBlockLoader(boolean columnReader) throws IOException {
+    private void testBlockLoader(boolean syntheticSource, boolean columnReader) throws IOException {
         SyntheticSourceExample example = syntheticSourceSupport(false).example(5);
-        MapperService mapper = createMapperService(syntheticSourceMapping(b -> { // TODO randomly use syntheticSourceMapping or normal
-            b.startObject("field");
-            example.mapping().accept(b);
-            b.endObject();
-        }));
+        // TODO when synthetic _source is disabled disable doc values.
```

Comment:
Most randomly disable doc_values and turn on store so we're ok here.

Still, it'd be nice to be able to make these with an explicit signal that we're not running in synthetic _source mode.